Source/WebCore/page/MemoryRelease.cpp

```diff
@@ -84,7 +84,12 @@ static void releaseNoncriticalMemory(MaintainMemoryCache maintainMemoryCache)
     GlyphDisplayListCache::singleton().clear();
     SelectorQueryCache::singleton().clear();
 
-    for (auto& document : Document::allDocuments()) {
+    auto allDocuments = Document::allDocuments();
+    auto protectedDocuments = WTF::map(allDocuments, [](auto& document) -> Ref<Document> {
+        return document.get();
+    });
+
+    for (auto& document : protectedDocuments) {
```

Comment:
Can't we just use `for(Ref document : allDocuments())`

Reply:
No because if that protected ref is the last ref to the document it will go out of scope at the end of the loop iteration and potentially mutate the allDocumentsMap.